cli/go-gh
# Replace duplicated IsEnterprise and IsTenancy functions in api pkg with auth pkg

Related to github/cli#577

## Changes

In attempting to use `go-gh` in `cli/cli` for `IsEnterprise` and `IsTenancy`, I got bamboozled by the duplicate code, here. This removes the duplicated, unexported `isEnterprise` and `isTenancy` functions from the `api` package and replaces them with the exported `IsEnterprise` and `IsTenancy` packages from `auth`.

## Testing

1. Pull down branch
2. Run tests

Note, I've also tested these in cli/cli by running `go mod edit --replace=github.com/cli/go-gh=<localPath>/go-gh` and swapped these functions in. Everything there seems to be working as expected, too. I ran all the tests and did some `gh auth` testing, just to be sure
